src/AutoRest.CSharp/MgmtTest/Generation/Mock/MgmtMockTestBaseWriter.cs

```diff
@@ -44,6 +48,75 @@ public virtual void Write()
+            _writer.UseNamespace("System");
+            _writer.UseNamespace("Azure.Identity");
+            _writer.UseNamespace("System.Threading.Tasks");
+
+            _writer.Line($"#region Parameter decalarations");
+            _writer.Line($"// Below parameters are extracted from swagger example files, please use your own values to execute the API lively.");
+            _writer.Line($"ArmClient GetArmClient() => new ArmClient(new DefaultAzureCredential());");
+            //RequestPath requestPath = testCase.Carrier switch
+            //{
+            //    ResourceCollection parentCollection => parentCollection.RequestPath,
+            //    Resource parentResource => parentResource.RequestPath,
+            //    MgmtExtensions parentExtension => parentExtension.ContextualPath,
+            //    _ => throw new InvalidOperationException($"Unknown parent {testCase.Carrier.GetType()}"),
+            //};
+            RequestPath requestPath = testCase.RequestPath;
+            foreach (var fs in testCase.ComposeResourceIdentifierParameterValues(requestPath, true))
+            {
+                var s = $"{fs}";
+                // s = s.Substring(1, s.Length - 2);
+                if (s == "subscriptionId")
+                {
+                    _writer.Line($"var subscriptionId = Environment.GetEnvironmentVariable(\"SUBSCRIPTION_ID\") ?? \"00000000-0000-0000-0000-000000000000\";");
+                }
+                else
+                {
+                    object? rawValue = null;
+                    string? description = null;
+                    foreach (var param in testCase.Example.AllParameters)
+                    {
+                        if (param.Parameter.Language.Default.SerializedName == s)
+                        {
+                            rawValue = param.ExampleValue.RawValue;
+                            description = param.Parameter.Summary ?? param.Parameter.Language.Default.Description;
+                        }
+                    }
+                    _writer.Append($"var {s} = ");
+                    _writer.AppendRawValue(rawValue?.GetType() ?? typeof(object), rawValue);
+                    if (description is not null)
+                    {
+                        _writer.Line($";  // {description}");
+                    }
+                    else
+                    {
+                        _writer.Line($";");
+                    }
+                    parameterVariables.Add(s);
+                }
+            }
+            _writer.Line($"#endregion");
+            _writer.Line($"");
+            _writer.Line($"#region API invocation");
+            _writer.Line($"// api-version: {testCase.RestOperation.Operation.ApiVersions.FirstOrDefault().Version}");
+            _writer.Line($"// x-ms-original-file: {testCase.Example.OriginalFile}");
+            WriteTestMethodBody(testCase);
+            _writer.Line($"#endregion");
+            generatingSample = false;
```

Comment:
This flag is really not a good desgin. We should have a separate class for writing the samples, and the class will automatically becomes the flag, therefore we do not need to maintain the state of this flag anymore